File: Server/src/main/java/server/AgencyService.java
package server;

import model.domain.Agency;
import repository.IAgencyRepo;

import java.util.Collection;

public class AgencyService {
    private final IAgencyRepo repo;

    public AgencyService(IAgencyRepo repo) {
        this.repo = repo;
    }

    public Collection<Agency> getAllAgencies() {
        return repo.getAll();
    }

    public Agency getAgency(Agency agency) { return repo.getAgency(agency.getName(), agency.getPassword()); }
}

File: Server/server/AgencyService.cs
using System;
using System.Collections.Generic;
using Model;
using Persistence;

namespace Server.server
{
    public class AgencyService
    {
        private readonly IAgencyRepo _repo;
        public AgencyService(IAgencyRepo repo)
        {
            _repo = repo;
        }
        public IEnumerable<Agency> GetAll() => _repo.GetAll();

        public Agency Get(Agency agency) => _repo.Get(agency.Name, agency.Password);
    }
}